### PR TITLE
Use version.txt to generate (pseudo-)version information

### PR DESCRIPTION
**Pull request by 3u13r**
### Proposed change(s)

Instead of walking the git tree to the first label found, we rely on a check-in `version.txt`. This removes the need to check-out the git history and the tags to build the binaries with correct stamps/version information.

build-os-image run: https://github.com/edgelesssys/constellation/actions/runs/4407760616
miniconstellation test this change is needed for: https://github.com/edgelesssys/constellation/actions/runs/4408501059/jobs/7723591433

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: hack/pseudo-version/pseudo-version.go [hunk 1]
```diff
@@ -91,9 +94,11 @@ func main() {
...
 		fmt.Println(parsedBranch)
-	case *printReleaseVersion:
-		fmt.Println(strings.TrimPrefix(rawBranch, "release/"))
 	default:
-		fmt.Println(version)
+		if !strings.Contains(*base, "pre") {
+			fmt.Println(*base) // v2.7.0
+		} else {
+			fmt.Println(version) // 2.7.0-pre.0.20230313121936-bab76e8a9acf
+		}
```
Comment (malt3) [inline hack/pseudo-version/pseudo-version.go:102]: not sure i like the example but it does kind of help to understand what will be printed so feel free to keep it.

File: hack/pseudo-version/pseudo-version.go [hunk 2]
```diff
@@ -44,19 +45,21 @@ func main() {
...
 	if *base == "" {
-		_, versionTag, err := gitc.FirstParentWithVersionTag()
+		rootPath, err := gitc.Path()
 		if err != nil {
-			log.With(zap.Error(err)).Warnf("Failed to find base version. Using default.")
-			versionTag = ""
+			log.With(zap.Error(err)).Fatalf("Failed to get git root path. Using default base version.")
+			//
+		} else {
+			versionTag, err := os.ReadFile(path.Join(rootPath, "/version.txt"))
```
Comment (malt3) [inline hack/pseudo-version/pseudo-version.go:54]: nit for 🪟

```suggestion
			versionTag, err := os.ReadFile(filepath.Join(rootPath, "version.txt"))
```

**Review by malt3 — APPROVED** [review]
thanks for making our builds more reproducible

File: hack/pseudo-version/pseudo-version.go (hunk 2 above)
Comment (Nirusu) [reply hack/pseudo-version/pseudo-version.go:54]: [Windows does not care whether you use `/` or `\`](https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats#canonicalize-separators) ;)
`C:/Windows\\system32/cmd.exe` for example is perfectly acceptable as path and works fine.

Though yes, `filepath` is better.

File: hack/pseudo-version/pseudo-version.go (hunk 1 above)
Comment (3u13r) [reply hack/pseudo-version/pseudo-version.go:102]: I think I know what you mean. Let me add a bit more verbosity here.

Comment (malt3) [reply hack/pseudo-version/pseudo-version.go]: Thanks

**Pull request merged**